pkg/ruler/notifier.go

```diff
@@ -158,14 +163,42 @@ func amConfigWithSD(rulerConfig *Config, url *url.URL, sdConfig discovery.Config
 
 	// Whether to use TLS or not.
 	if rulerConfig.Notifier.TLSEnabled {
-		amConfig.HTTPClientConfig.TLSConfig = config_util.TLSConfig{
-			CAFile:             rulerConfig.Notifier.TLS.CAPath,
-			CertFile:           rulerConfig.Notifier.TLS.CertPath,
-			KeyFile:            rulerConfig.Notifier.TLS.KeyPath,
-			InsecureSkipVerify: rulerConfig.Notifier.TLS.InsecureSkipVerify,
-			ServerName:         rulerConfig.Notifier.TLS.ServerName,
+		if rulerConfig.Notifier.TLS.Reader == nil || reflect.ValueOf(rulerConfig.Notifier.TLS.Reader).IsNil() {
```

Comment:
Can we *not* do `reflect.ValueOf(rulerConfig.Notifier.TLS.Reader).IsNil()` part?